File: e-polling/lib/models/candidate.dart
class Candidate {
  String id;
  String name;
  String voting;
  String createdAt;
  String updatedAt;
  String? description;
  String? image;
  int votes;

  Candidate({
    required this.id,
    required this.name,
    required this.voting,
    required this.createdAt,
    required this.updatedAt,
    this.description,
    this.image,
    required this.votes,
  });

  static Candidate fromJson(Map<String, dynamic> json) {
    return Candidate(
      id: json['_id'],
      name: json['name'],
      voting: json['voting'],
      createdAt: json['createdAt'],
      updatedAt: json['updatedAt'],
      description: json['description'],
      image: json['image'],
      votes: json['votes'],
    );
  }
}

File: e-polling-backend/src/models/candidate.js
const mongoose = require('mongoose');
const { formattedImagePath } = require('../utils/helpers');
const Vote = require('./vote');

const candidateSchema = mongoose.Schema(
  {
    name: {
      type: String,
      required: true,
      trim: true,
    },
    description: {
      type: String,
      trim: true,
    },
    image: {
      type: String,
    },
    voting: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Voting',
      required: true,
    },
  },
  {
    timestamps: true,
  }
);

candidateSchema.methods.formatted = async function (req) {

    return {
      ...this.toObject(),
      votes: await Vote.countDocuments({ candidate: this._id }),
      image: !this.image ? undefined : formattedImagePath(req, this.image),
    };

}

const Candidate = mongoose.model('Candidate', candidateSchema);
module.exports = Candidate;